tests/bluetooth/tester/src/btp_vcp.c

```diff
@@ -0,0 +1,544 @@
+/* vcp.c - Bluetooth VCP Tester */
+
+/*
+ * Copyright (c) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/bluetooth/bluetooth.h>
+
+#include <zephyr/bluetooth/testing.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/bluetooth/audio/vcp.h>
+#include <zephyr/bluetooth/audio/aics.h>
+#include "zephyr/bluetooth/audio/vocs.h"
+#include <app_keys.h>
+
+#include <zephyr/logging/log.h>
+#define LOG_MODULE_NAME bttester_vcp
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include "btp/btp.h"
+
+#define CONTROLLER_INDEX 0
+
+struct bt_vcp_vol_rend_register_param vcp_register_param;
+struct bt_vcp_included included;
+static bool g_aics_active = 1;
+
+static void set_register_params(uint8_t gain_mode);
+
+/* Volume Control Service */
+static void vcs_supported_commands(uint8_t *data, uint16_t len)
+{
+	struct net_buf_simple *buf = NET_BUF_SIMPLE(BTP_DATA_MAX_SIZE);
+
+	net_buf_simple_init(buf, 0);
+	net_buf_simple_add_u8(buf, VCS_READ_SUPPORTED_COMMANDS);
+	net_buf_simple_add_u8(buf, VCS_INIT);
+	net_buf_simple_add_u8(buf, VCS_SET_VOL);
+	net_buf_simple_add_u8(buf, VCS_VOL_UP);
+	net_buf_simple_add_u8(buf, VCS_VOL_DOWN);
+	net_buf_simple_add_u8(buf, VCS_MUTE);
+	net_buf_simple_add_u8(buf, VCS_UNMUTE);
+
+	tester_send(BTP_SERVICE_ID_VCS, VCS_READ_SUPPORTED_COMMANDS,
+		    CONTROLLER_INDEX, buf->data, buf->len);
+}
+
+static void set_volume(uint8_t *data, uint16_t len)
+{
+	const struct vcs_set_vol_cmd *cmd = (void *) data;
+	uint8_t status = BTP_STATUS_SUCCESS;
+	uint32_t volume;
+	int err;
+
+	volume = sys_le32_to_cpu(cmd->volume);
+
+	LOG_DBG("volume 0x%04x", volume);
```

Comment:
Is the volume field supposed to be 32-bit? 
The actual volume field is 8-bit. 
The volume state is however 24-bit. 

In any case, logging 16-bit here doesn't really make any sense

Reply:
Done